relayer/pkg/chainlink/txm/txm.go

```diff
@@ -55,9 +65,23 @@ type starktxm struct {
 	client       *utils.LazyLoad[*starknet.Client]
 	feederClient *utils.LazyLoad[*starknet.FeederClient]
 	accountStore *AccountStore
+	metrics      TxMetrics
+	chainID      string
+
+	// Track broadcast times for confirmation duration metrics
+	broadcastTimes sync.Map // map[string]time.Time (tx hash -> broadcast time)
+
+	// Track retry attempts for max attempts detection
+	retryAttempts sync.Map // map[string]int (tx hash -> attempt count)
```

Comment:
I prefer using nonces instead of hashes to uniquely identify a transaction for broadcastTimes and retryAttempts. EVM txmv2 stores/fetches by nonce (though there is an option to fetch by txm attempt, but that's because you can associated one tx to many tx attempts in EVM txmv2). Since starknet txm also is nonce-centric we should do the same thing here.

By using nonce, these intermediate data structures won't have abandoned/stale txs in the map since they will be overwritten if we resync the nonce in the case of a reorg or sequencer degradation like we saw recently. 

Tracking by nonce also helps because if we tag the metrics by nonce and see if a nonce is being broadcasted multiple times but has no confirmation.